src/http/headers.cr

```diff
@@ -9,13 +9,27 @@ struct HTTP::Headers
   record Key, name : String do
     forward_missing_to @name
 
-    def hash
-      h = 0
-      name.each_byte do |c|
-        c = normalize_byte(c)
-        h = 31 * h + c
+    def hash(hasher)
+      # hash 4 bytes at a row
+      c = 0
+      bytesize = name.bytesize
+      slice = name.to_slice
+      hasher.raw(bytesize)
+      while c + 4 < bytesize
```

Comment:
Slice looks to allocate Array. Am I mistaken?
Computing hash should not do heap allocation.

Where is `step` defined? I didn't find it in 'int.cr' (while `upto` were there) and gave up.
Ah, I see, it is in `number.cr`. My bad. I'll fix it.